src/nnet3/nnet-training.cc

```diff
@@ -222,11 +224,10 @@ void NnetTrainer::PrintMaxChangeStats() const {
 void ObjectiveFunctionInfo::UpdateStats(
     const std::string &output_name,
     int32 minibatches_per_phase,
-    int32 minibatch_counter,
```

Comment:
I think you should revert this change and do it another way if possible.  It's too likely to cause problems elsewhere (e.g. merge conflicts with adversarial training).  When you call it with `num_minibatches_processed_++`, can't you just delay incrementing num_minibatches_processed_ until later on?